source/common/upstream/upstream_impl.cc

```diff
@@ -975,6 +978,20 @@ ClusterInfoImpl::ClusterInfoImpl(
         factory.createFilterFactoryFromProto(*message, *factory_context_);
     filter_factories_.push_back(std::move(callback));
   }
+
+  if (http_protocol_options_ && !http_protocol_options_->http_filters_.empty()) {
+    std::shared_ptr<Http::UpstreamFilterConfigProviderManager> filter_config_provider_manager =
+        Http::FilterChainUtility::createSingletonUpstreamFilterConfigProviderManager(
+            server_factory_context_);
+    Http::FilterChainUtility::FiltersList http_filters = http_protocol_options_->http_filters_;
+
+    std::string prefix = stats_scope_->symbolTable().toString(stats_scope_->prefix());
+    Http::FilterChainHelper<Server::Configuration::ServerFactoryContext,
+                            Server::Configuration::UpstreamHttpFilterConfigFactory>
+        helper(*filter_config_provider_manager, server_factory_context_, server_factory_context_,
+               prefix);
+    helper.processFilters(http_filters, "http", "http", http_filter_factories_, false);
+  }
```

Comment:
Does this print any logs when it's building the chain similar to how the downstream logs print? Might be useful for debugging / just letting people know that the config was correct? It also occurs to me that obviously with the downstream upstream/split there is no guarantee these will ever get called if we don't end up doing HTTP on the cluster but I don't think there is much we can do about that, much like the other settings I guess. Mostly just thinking about how to let people know what they configured is correct and working. Fine to have a TODO / follow up in a future PR.